test/ParserTests.hs

```diff
@@ -571,6 +575,31 @@ testTypeSynCannotBeItsOwnValue = TestCase (
   False
   (isRight $ parseAll (parseGame []) "" "game E\ntype AB={AB}"))
 
+testTypeExtSyn1 :: Test
+testTypeExtSyn1 = TestCase (
+  assertEqual "Test that a type can be extended by an etype synonym"
+  True
+  (isRight $ parseAll (parseGame []) "" "game E\ntype T1 = {A}\ntype T2 = Int & T1"))
+
+testTypeExtSyn2 :: Test
+testTypeExtSyn2 = TestCase (
+  assertEqual "Test that a type can be extended by an etype synonym"
+  True
+  (isRight $ parseAll (parseGame []) "" "game E\ntype T1 = {A}\ntype T2 = T1 & {B}"))
+
+-- todo: remove the testTypeExtLimitation tests when it is moved to the type checker
+testTypeExtLimitation1 :: Test
+testTypeExtLimitation1 = TestCase (
+  assertEqual "Test that a type cannot be extended by a non etype"
+  False
+  (isRight $ parseAll (parseGame []) "" "game E\ntype T1 = Int & {A}\ntype T2 = Int & T1"))
+
+testTypeExtLimitation2 :: Test
+testTypeExtLimitation2 = TestCase (
+  assertEqual "Test that a type cannot be extended by a non etype"
+  False
+  (isRight $ parseAll (parseGame []) "" "game E\ntype T1 = Board\ntype T2 = Int & T1"))
+
```

Comment:
Tests look good 👍 . Personally I swear we could do this before, but I guess it we never set it up 🤔 .